### DISCUSSION TITLE
Vue Combobox "by" prop doesn't seem to work

### DISCUSSION THREAD

**Discussion by AnnaYuS** [tailwindlabs/headlessui#1682] [Help] [answered]
I'm trying to make combobox work with a string as a v-model value. The model value is a string which is an id of one of the objects in the array of options.
I've noticed that a new prop to match the model value and the object in the options array [has been added recently](https://github.com/tailwindlabs/headlessui/pull/1482).  To my understanding this should make the scoped prop of ComboboxOption `select` work in this case for example: https://stackblitz.com/edit/vue-aqjzwb?file=src/components/HelloWorld.vue but i cannot make it work. Maybe there is a problem in my implementation or maybe it's a bug because i've noticed that in Vue dev tools `by` is listed under `attrs`, not `props`.
I would really appreciate some help 🙏

**Comment by RobinMalfait** [comment, marked as answer]
Hey! Thank you for your question!
Much appreciated! 🙏

Right now the `by` prop is not released yet, you can checkout the CHANGELOG.md file to know what is released and what isn't released: https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-vue/CHANGELOG.md

That said, everything that gets merged to the main branch becomes available on npm using the `insiders` tag. This means that if you want to use the `by` prop that you should use the insiders build right now.

Note, the `insiders` build is not following semver. More info: https://github.com/tailwindlabs/headlessui#installing-the-insiders-version

Here is an updated stackblitz.com link for you using the insiders build: https://stackblitz.com/edit/vue-mzzooa?file=src%2Fcomponents%2FHelloWorld.vue

Hope this helps!

**Reply by AnnaYuS** [reply]
Ah, got confused there. It's good to know about the insiders build 👍  And it works as expected! 
Thank you!

**Reply by thoresuenert** [reply]
@RobinMalfait 
We have a situation where we call an api on searchQuery changes.
In our situation we don't have equal objects.

We now have the situation, where `{id: 1}` is selected.
The api search will give us the same option but different object `{id: 1}`.
In this situation we cannot remove this object.

I think the `selectOption` function relies on strict checking because of `indexOf`.
```js
copy.indexOf(raw)
```
This results in `never finding` the right object which will result in duplicate entries.
The Solution should use the provided `by=compare` function to find the right object in the selected state.

how to report this issue correctly?

**Reply by RobinMalfait** [reply]
Aha, thanks for letting me know! Will fix this.
